### DISCUSSION TITLE
Is there any way to use image generation models available on Shuttle AI?

### DISCUSSION THREAD

**Discussion by kneelesh48** [danny-avila/LibreChat#2186] [Feature Requests & Suggestions]
The chat models work, but selecting image generation models from the drop-down and sending a chat message results in api error as shown in the screenshot below

![Screenshot_2024-03-24-09-38-17-36_4d38fce200f96aeac5e860e739312e76](https://github.com/danny-avila/LibreChat/assets/25109737/15accdf3-7494-4645-98f1-dba7fe1595d5)

**Comment by fuegovic** [comment]
Maybe, but not easily (this has not been tested and might not work at all)

Image generation only works with plugins currently. You might be able to send a request to the dall-e / dall-e-3 models if you configure the following .env variables to point to ShuttleAI:

You can try setting either
```yaml 
DALLE_REVERSE_PROXY=
```
or:
```yaml
DALLE3_BASEURL=
DALLE2_BASEURL=
```
Then install the DALL-E and/or DALL-E-3 plugins and test

Another thing to consider is that the plugin endpoint is using OpenAI for the thought process. If you don't have OpenAI access you would also need to configure ShuttleAI to hijack the OpenAI endpoint with this .env variable:

```yaml
OPENAI_REVERSE_PROXY=
```

Even after completing all these steps, success is not guaranteed and the models would be limited to `dall-e` and `dall-e-3`.

---

@danny-avila 
Would it be possible to introduce endpoints dedicated solely to image generation in LibreChat? Although we currently support numerous endpoints capable of image generation, there isn’t a straightforward method to use the feature.
